activerecord/lib/active_record/associations/join_dependency/join_association.rb

```diff
@@ -38,41 +38,39 @@ def join_constraints(foreign_table, foreign_klass, join_type, alias_tracker)
             chain << [reflection, table]
           end
 
-          base_klass.with_connection do |connection|
```

Comment:
As a bonus, we no longer need to checkout the connection here. This was a big smell.